File: bialydunajec-registrations/presentation/src/main/kotlin/org/bialydunajec/registrations/presentation/rest/v1/admin/request/UpdateCottageRequest.kt
package org.bialydunajec.registrations.presentation.rest.v1.admin.request

import org.bialydunajec.ddd.application.base.query.api.dto.PlaceDto
import org.bialydunajec.ddd.domain.base.validation.constraints.NullOrNotBlank
import org.bialydunajec.registrations.application.dto.BankTransferDetailsDto
import org.bialydunajec.registrations.application.dto.CampersLimitationsDto
import org.bialydunajec.registrations.application.dto.CottageSpaceDto
import javax.validation.Valid
import javax.validation.constraints.NotBlank

data class UpdateCottageRequest(
        @field:NotBlank
        val name: String,
        @field:NullOrNotBlank
        val logoImageUrl: String?,
        @field:NullOrNotBlank
        val buildingPhotoUrl: String?,
        @field:Valid
        val place: PlaceDto?,
        @field:Valid
        val cottageSpace: CottageSpaceDto?,
        @field:Valid
        val campersLimitations: CampersLimitationsDto?,
        @field:Valid
        val bankTransferDetails: BankTransferDetailsDto?
)

File: bialydunajec-adminpanel-frontend/projects/bialydunajec-admin/src/app/camp-registrations/service/rest/request/update-cottage.request.ts
import {PlaceDto} from '../../../../shared/service/rest/dto/place.dto';
import {CottageSpaceDto} from '../dto/cottage-space.dto';
import {CampersLimitationsDto} from '../dto/camper-limitations.dto';
import {BankTransferDetailsDto} from '../dto/bank-transfer-details.dto';

export class UpdateCottageRequest {
  campRegistrationsEditionId: string;
  cottageType: string;
  academicMinistryId?: string;
  name: string;
  logoImageUrl?: string;
  buildingPhotoUrl?: string;
  place?: PlaceDto;
  cottageSpace: CottageSpaceDto;
  campersLimitations?: CampersLimitationsDto;
  bankTransferDetails?: BankTransferDetailsDto;
  cottageState: string;

  constructor(campRegistrationsEditionId: string, cottageType: string, academicMinistryId: string, name: string, logoImageUrl: string, buildingPhotoUrl: string, place: PlaceDto, cottageSpace: CottageSpaceDto, campersLimitations: CampersLimitationsDto, bankTransferDetails: BankTransferDetailsDto, cottageState: string) {
    this.campRegistrationsEditionId = campRegistrationsEditionId;
    this.cottageType = cottageType;
    this.academicMinistryId = academicMinistryId;
    this.name = name;
    this.logoImageUrl = logoImageUrl;
    this.buildingPhotoUrl = buildingPhotoUrl;
    this.place = place;
    this.cottageSpace = cottageSpace;
    this.campersLimitations = campersLimitations;
    this.bankTransferDetails = bankTransferDetails;
    this.cottageState = cottageState;
  }
}
